lib/viewing/RenderableView.js

```diff
@@ -58,10 +54,38 @@ var RenderableView = _.extend({},
             }.bind(this));
         },
 
+        generateViewPlaceholders: function() {
+            var unplacedCount = this.childViewCount() - this.unrenderedChildViewCount();
+
+            // [wawjr3d] Do this before return so data.views is always an object
+            this.viewPlaceholders = new Array(unplacedCount);
+
+            if (!unplacedCount) {
```

Comment:
to set this.viewPlaceholders to an empty array by default we'd need to do that in a constructor. this object is a mixin without a constructor. i think we should discuss changing how mixins are used to create Views. i think that change is a refactoring that belongs in its own PR